test.py

```diff
@@ -216,6 +216,43 @@ def test_get_fen(self):
         result = sf.get_fen("makruk", fen, [], False, False, True)
         self.assertEqual(result, fen)
 
+        # makruk piece honor counting
+        fen = "8/3k4/8/2K1S1P1/8/8/8/8 w - - 0 50"
+        moves = ["g5g6M"]
+        result = sf.get_fen("makruk", fen, moves, False, False, True)
+        self.assertEqual(result, "8/3k4/6M~1/2K1S3/8/8/8/8 b - 88 8 50");
+
+        fen = "8/2K3k1/5m2/4S1S1/8/8/8/8 w - 128 97 70"
+        moves = ["e5f6"]
+        result = sf.get_fen("makruk", fen, moves, False, False, True)
+        self.assertEqual(result, "8/2K3k1/5S2/6S1/8/8/8/8 b - 44 8 70");
+
+        # makruk board honor counting
+        fen = "3k4/2m5/8/4MP2/3KS3/8/8/8 w - - 0 1"
+        moves = ["f5f6M"]
+        result = sf.get_fen("makruk", fen, moves, False, False, True)
+        self.assertEqual(result, "3k4/2m5/5M~2/4M3/3KS3/8/8/8 b - 128 0 1");
+
+        fen = "3k4/2m5/5M~2/4M3/3KS3/8/8/8 w - 128 0 33"
+        moves = ["d4d5"]
+        result = sf.get_fen("makruk", fen, moves, False, False, True)
+        self.assertEqual(result, "3k4/2m5/5M~2/3KM3/4S3/8/8/8 b - 128 1 33");
+
+        fen = "3k4/2m5/5M~2/4M3/3KS3/8/8/8 w - 128 0 33"
+        moves = ["d4d5"]
+        result = sf.get_fen("makruk", fen, moves, False, False, True, 0)
+        self.assertEqual(result, "3k4/2m5/5M~2/3KM3/4S3/8/8/8 b - 128 1 33");
+
```

Comment:
It would be good to also cover the above mentioned scenario of gamePly < countingPly (with countStarted == 0) in the tests.